R/as_gt.R

```diff
@@ -0,0 +1,144 @@
+#  Copyright (c) 2024 Merck & Co., Inc., Rahway, NJ, USA and its affiliates.
+#  All rights reserved.
+#
+#  This file is part of the simtrial program.
+#
+#  simtrial is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' Convert summary table to a gt object
+#'
+#' @param x A summary object of a fixed or group sequential design.
+#' @param ... Additional arguments (not used).
+#'
+#' @return A gt table.
+#'
+#' @export
+as_gt <- function(x, ...) {
+  UseMethod("as_gt", x)
+}
+
+
+#' @param x A object returned by [summary()].
+#' @param title Title of the gt table.
+#' @param subtitle Subtitle of the gt table.
+#' @param ... Additional parameters (not used).
+#'
+#' @return A gt table summarizing the simulation results.
+#' @export
+#' @rdname as_gt
+#'
+#' @examples
+#'
+#' # Parameters for enrollment
+#' enroll_rampup_duration <- 4 # Duration for enrollment ramp up
+#' enroll_duration <- 16 # Total enrollment duration
+#' enroll_rate <- gsDesign2::define_enroll_rate(
+#'   duration = c(
+#'     enroll_rampup_duration, enroll_duration - enroll_rampup_duration),
+#'  rate = c(10, 30))
+#'
+#' # Parameters for treatment effect
+#' delay_effect_duration <- 3 # Delay treatment effect in months
+#' median_ctrl <- 9 # Survival median of the control arm
+#' median_exp <- c(9, 14) # Survival median of the experimental arm
+#' dropout_rate <- 0.001
+#' fail_rate <- gsDesign2::define_fail_rate(
+#'   duration = c(delay_effect_duration, 100),
+#'   fail_rate = log(2) / median_ctrl,
+#'   hr = median_ctrl / median_exp,
+#'   dropout_rate = dropout_rate)
+#'
+#' # Other related parameters
+#' alpha <- 0.025 # Type I error
+#' beta <- 0.1 # Type II error
+#' ratio <- 1 # Randomization ratio (experimental:control)
+#'
+#' # Build a one-sided group sequential design
+#' design <- gsDesign2::gs_design_ahr(
+#'   enroll_rate = enroll_rate, fail_rate = fail_rate,
+#'   ratio = ratio, alpha = alpha, beta = beta,
+#'   analysis_time = c(12, 24, 36),
+#'   upper = gsDesign2::gs_spending_bound,
+#'   upar = list(sf = gsDesign::sfLDOF, total_spend = alpha),
+#'   lower = gsDesign2::gs_b,
+#'   lpar = rep(-Inf, 3))
+#'
+#' # Define cuttings of 2 IAs and 1 FA
+#' ia1_cut <- simtrial::create_cut(target_event_overall = ceiling(design$analysis$event[1]))
```

Comment:
We shouldn't need to prefix with `simtrial::` when these examples are part of the {simtrial} package

Reply:
Updated